pkg/sql/create_stats.go

```diff
@@ -162,25 +161,33 @@ func (n *createStatsNode) startJob(ctx context.Context) error {
 				log.Warningf(ctx, "failed to cleanup StartableJob: %v", cleanupErr)
 			}
 		}
-		return err
+		return nil, err
 	}
 	if err := job.Start(ctx); err != nil {
-		return err
+		return nil, err
```

Comment:
perhaps a question for the Queries team, but why so we want to start the job here, rather than in `Next()`? wouldn't that let us avoid the need for async tasks?